### ISSUE TITLE
FN — a component that returns `null` on every path is not detected at all

### ISSUE THREAD

**Issue by rboudrouss** [rboudrouss/reactant-analyzer#122]
Found by the blind-wish-list campaign (two independent agents hit it), then reduced.

## Repro

```tsx
// A.tsx — components_analyzed: 0, zero findings
export function NullComp({ ms }) {
  const [n, setN] = useState(0);
  useEffect(() => { setInterval(() => { setN(n + 1); }, ms); }, []);
  return null;
}
```

```tsx
// B.tsx — components_analyzed: 1, four findings including an Error-tier stale-closure
export function JsxComp({ ms }) {
  const [n, setN] = useState(0);
  useEffect(() => { setInterval(() => { setN(n + 1); }, ms); }, []);
  return <i />;
}
```

Measured:

```
A.tsx  components_analyzed: 0   (no diagnostics)
B.tsx  components_analyzed: 1   missing-cleanup, missing-deps ×2, stale-closure
```

A component with a `return null` on *some* path is fine — a third fixture with
`if (!open) return null; return <i/>;` is analysed normally. It is the
all-paths-`null` component that vanishes.

## Cause

`component_detector::is_component` (`src/lowering/component_detector.rs`) accepts
a capitalised non-`use` function on exactly two positive signals: some return
path yields JSX (`jsx_detect::body_returns_jsx`), or the return type is a
TS component annotation. A component that always returns `null` — and is not
TS-annotated — matches neither, so it is never lowered, never registered, and
never analysed.

## Why it matters

Null-rendering components are a normal React idiom, and they are *exactly* where
lifecycle bugs live: analytics and telemetry mounts, `useEffect`-only wrappers,
portal hosts, feature-flag gates, keyboard-shortcut registrars. The whole
`missing-cleanup` / `stale-closure` / `infinite-loop` family is silent on them,
with no Info to say so — the component simply is not in the report.

## Fix direction

Add a third positive signal to `is_component`: **a capitalised non-`use`
function that calls a React hook is a component.** That is the Rules of Hooks
read backwards — nothing but a component or a hook may call one, and rule 1
already excludes hooks by name — so it is a one-sentence rule rather than a
heuristic, and it needs only the AST already in hand at detection time.

Note the symmetry with `utility_detector`, which uses the same `body_returns_jsx`
predicate as a *negative* signal: a function that calls hooks must not be
classified as a utility either.

**Comment by rboudrouss** [comment]
Fixed by rule 4: a capitalised non-`use` function that calls a React hook is a component. Corpus effect is purely additive — +88 on twenty, +7 dub, +2 chakra-ui, +2 memos, nothing removed.
